A_Perfect_Dish_to_Share.py

```diff
@@ -0,0 +1,38 @@
+def perfect_dish_to_share():
+    """
+    A function that prints all the "Perfect dish to share" numbers.
+    :return: none
+    """
+    for item in search_for_all_perfect_dish():
+        print(item)
+
+
+def search_for_all_perfect_dish():
+    """
+    A generator function that search for all the "Perfect dish to share" number.
+    :return: The next "Perfect dish to share" number.
+    """
+    num = 1
+    while True:
+        if is_perfect_dish(num):
+            yield num
+        num += 1
+
+
+def is_perfect_dish(dish):
+    """
+    A functions that get an integer and return if the given number is a "Perfect dish to share"
```

Comment:
The function gets an integer and returns True if the given number is a "Perfect dish to share", else False.